mlrun/model_monitoring/db/tsdb/tdengine/schemas.py

```diff
@@ -125,33 +125,66 @@ def _get_subtables_query(
     @staticmethod
     def _get_records_query(
         table: str,
-        start: str,
-        end: str,
+        start: datetime,
+        end: datetime,
         columns_to_filter: list[str] = None,
-        filter_query: str = "",
+        filter_query: Optional[str] = None,
+        interval: Optional[str] = None,
+        limit: int = 0,
+        agg_funcs: Optional[list] = None,
+        sliding_window_step: Optional[str] = None,
         timestamp_column: str = "time",
         database: str = _MODEL_MONITORING_DATABASE,
     ) -> str:
+        if agg_funcs and not columns_to_filter:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "`columns_to_filter` must be provided when using aggregate functions"
+            )
+
+        # if aggregate function or interval is provided, the other must be provided as well
+        if interval and not agg_funcs:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "`agg_funcs` must be provided when using interval"
+            )
+
+        if sliding_window_step and not interval:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "`interval` must be provided when using sliding window"
+            )
+
         with StringIO() as query:
             query.write("SELECT ")
-            if columns_to_filter:
+            if interval:
+                query.write("_wstart, _wend, ")
+            if agg_funcs:
+                query.write(
+                    ", ".join(
+                        [f"{a}({col})" for a in agg_funcs for col in columns_to_filter]
+                    )
+                )
+            elif columns_to_filter:
                 query.write(", ".join(columns_to_filter))
             else:
                 query.write("*")
-            query.write(f" from {database}.{table}")
+            query.write(f" FROM {database}.{table}")
 
             if any([filter_query, start, end]):
-                query.write(" where ")
+                query.write(" WHERE ")
                 if filter_query:
-                    query.write(f"{filter_query} and ")
+                    query.write(f"{filter_query} AND ")
                 if start:
-                    query.write(f"{timestamp_column} >= '{start}'" + " and ")
+                    query.write(f"{timestamp_column} >= '{start}'" + " AND ")
                 if end:
                     query.write(f"{timestamp_column} <= '{end}'")
-                full_query = query.getvalue()
-                if full_query.endswith(" and "):
-                    full_query = full_query[:-5]
-            return full_query + ";"
+            if limit or interval:
+                if interval:
+                    query.write(f" INTERVAL({interval})")
+                if sliding_window_step:
+                    query.write(f" SLIDING({sliding_window_step})")
+                if limit:
+                    query.write(f" LIMIT {limit}")
```

Comment:
Hm, do we need the outer if?

Reply:
good catch, I guess not because we are already handling "invalid" use cases before that (e.g. sliding window without interval).